calculator.rb

```diff
@@ -0,0 +1,64 @@
+# prints welcome message
+puts "Welcome to the Calculator! Which operator would you like to use?"
+puts "1. add(+)"
+puts "2. subtract(-)"
+puts "3. multiply(*)"
+puts "4. divide(/)"
+puts "5. compute exponent(^)"
+puts "6. compute modulo(%)"
+print "Please enter an operator (name or symbol): "
+
+# asks the user to enter an operator, prints error if invalid input
+oper = gets.chomp.downcase
+until ["+","add","-","subtract","*", "multiply","/","divide","^","compute exponent", "%", "compute modulo"].include?(oper) == true
+  print "You entered an invalid operator. Please enter a new operator: "
+  oper = gets.chomp
+end
+
+# asks the user to enter two numbers, prints error if invalid input
+print "Please enter your first number: "
+num1 = gets.chomp
+num1_test = num1.to_i
+while num1_test.to_s != num1
+  print "You entered an invalid number, please enter your first number again: "
+  num1 = gets.chomp
+  num1_test = num1.to_i
+end
+num1 = num1.to_i
+
+print "Please enter your second number: "
+num2 = gets.chomp
+num2_test = num2.to_i
+while num2_test.to_s != num2
+  print "You entered an invalid number, please enter your second number again: "
+  num2 = gets.chomp
+  num2_test = num2.to_i
```

Comment:
You noted this in your reflection, but you should convert to floats so that it handles float division correctly (5/6 should not equal 0)